src/App.js

```diff
@@ -16,32 +22,41 @@ import base from './styles/base'; // eslint-disable-line no-unused-vars
 import actions from './actions/index';
 import reducers from './reducers/index';
 
+const history = createBrowserHistory();
+const middleware = routerMiddleware(history);
+
 const sagaMiddleware = createSagaMiddleware();
+
 const store = createStore(
-  reducers,
-  {},
-  composeWithDevTools(applyMiddleware(sagaMiddleware)),
+  combineReducers({ ...reducers, router: routerReducer }),
+  composeWithDevTools(applyMiddleware(sagaMiddleware, middleware)),
 );
 sagaMiddleware.run(actions);
 
+const withDefault = component => () => (
```

Comment:
We introduced this, as otherwise the Default component is rendered before the routes are available (they are defined inside). I don't understand why exactly it doesn't work, but this fixes it.

Reply:
@lauriii Do you have any thoughts on this?

Reply:
```

Reply:
This is another way to do this, but like @dawehner pointed out, it's maybe a bit too complicated.

Reply:
I think the current approach works fine. Passing objects makes sense when you have more parameters (or you are likely to have).

Reply:
👏 👍